### PR TITLE
Keystore Generation & Decryption

### PR DESCRIPTION
**Pull request by crypdoughdoteth**
## Motivation

This is a functioning example of generating + encrypting a new keystore and decrypting a keystore using ethers-rs. This could particularly be helpful for applications that rely on local encrypted keys client side or server side.

### CODE REVIEW

File: examples/wallets/examples/keystores.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+use eyre::Result;
+
+
+#[tokio::main]
+async fn main() -> Result<()>{
+
+    dotenv().ok();
+
+    let api_key = std::env::var("API_KEY").expect("expected environmental variable");
+    let encryption_path = std::env::var("ENCYPTION_PATH").expect("expected environmental variable");
```
Comment (roynalnaruto) [inline examples/wallets/examples/keystores.rs:15]:
```suggestion
    let encryption_path = std::env::var("ENCRYPTION_PATH").expect("expected environmental variable");
```

File: examples/wallets/examples/keystores.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    let ws = Ws::connect(api_key)
+        .await
+        .expect("no connection");
+
+    println!("Is Connected: {}", ws.ready());
+
+    //generates a brand new keystore with key
+    //name your keystore file, set the path, set the password
+    //*NOTE* -- pls don't store your passwords in a plain text file, this is just an example 
+    let key_store = LocalWallet::new_keystore(encryption_path, &mut thread_rng(), &password, Some("my_encrypted_keys"))
```
Comment (roynalnaruto) [inline examples/wallets/examples/keystores.rs:28]:
```suggestion
    let (signing_key, _) = LocalWallet::new_keystore(encryption_path, &mut thread_rng(), &password, Some("my_encrypted_keys"))
```

File: examples/wallets/examples/keystores.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        .expect("key store fail");
+    
+    let (your_signing_key, _) = key_store;
```
Comment (roynalnaruto) [inline examples/wallets/examples/keystores.rs:31]: Then you can remove this line

**Review by roynalnaruto — COMMENTED** [review]
Minor suggestions and nitpicks

File: examples/wallets/examples/keystores.rs (hunk 3 above)
Comment (crypdoughdoteth) [reply examples/wallets/examples/keystores.rs]: Thank you! I merged your suggestions ❤️

Comment (roynalnaruto) [reply examples/wallets/examples/keystores.rs]: You might need to fix linting

Comment (crypdoughdoteth) [reply examples/wallets/examples/keystores.rs]: Linting worked locally, I think the bot doesn't like dotenv crate. So, I removed it and put placeholder values in

**Comment by roynalnaruto** [comment]
Actually for what's being done in this example, in my opinion the documentation above those methods should already be sufficient. But I believe @gakonst can decide if this example is a good to have!

**Review by gakonst — CHANGES_REQUESTED** [review]
Just looking at the docs.rs https://docs.rs/ethers/latest/ethers/signers/struct.Wallet.html#method.decrypt_keystore, I would rather that we had these here, instead of in the example. 

This example also isn't buildingm so my recommendation would be to move these to a doc comment in the struct/function:
1. Please add one of the public RPC urls we use
2. Please use a path that's not windows specific

**Comment by prestwich** [comment]
Closing as stale

**Pull request closed without merging**